lib/arborist/build-ideal-tree.js

```diff
@@ -130,6 +135,12 @@ module.exports = cls => class IdealTreeBuilder extends cls {
     this[_linkNodes] = new Set()
     this[_manifests] = new Map()
     this[_peerConflict] = null
+
+    // caches for cached realpath calls
+    const cwd = process.cwd()
+    // assume that the cwd is real enough for our purposes
+    this[_rpcache] = new Map([[ cwd, cwd ]])
```

Comment:
Since we already have a realpath cache on the Arborist object for use by Arborist.loadActual(), it'd be better for both of them to use `const _rpcache = Symbol.for('realpathCache')`, and only have one of them initialize it.  Otherwise we're doing double work and not caching as efficiently as we could be.